Documentation/ceph-cluster-crd.md

```diff
@@ -522,6 +522,11 @@ You can set resource requests/limits for Rook components through the [Resource R
 
 * `mon`: Set resource requests/limits for mons
 * `osd`: Set resource requests/limits for OSDs
+  This key applies for all OSDs regardless of their device classes. In case of need to apply resource requests/limits for OSDs with
+  particular device class use specific osd keys below.
+* `osd-hdd`: Set resource requests/limits for OSDs with HDD device class
```

Comment:
We can use our own `osd-foo` class too? If so, please document it, thanks! Same for the cluster.yaml

Reply:
Actually yes, as i dont see any validation for allowed device classes. But for now we have only "hdd,ssd and nvme" - do we need to explicitly say that it is possible to set "osd-foo" if no such device class?

Reply:
My point is that users might want to set their own device class to do some specific data placement and has a result allocate specific resources to those pod so yes I'd like to document that. Thanks

Reply:
What about this?
```suggestion
* `osd-<deviceClass>`: Set resource requests/limits for OSDs a particular device class, for example `osd-hdd` for HDDs or `osd-custom` for a custom device class.
```

Reply:
Or even better...
```suggestion
* `osd-<deviceClass>`: Set resource requests/limits for OSDs on a specific device class. Rook will automatically detect `hdd`, `ssd`, or `nvme` device classes. Custom device classes can also be set.
```

But we need better documentation about how to set the custom device class. Do you want to improve that in this PR, or we can find someone? I have an idea about it.

Reply:
If you don't mind, i'd like to have documentation change in different PR. As i see, we need update Documentation/ceph-cluster-crd.md and Documentation/ceph-pool-crd.md doc files to reflect that change. I can upload that change or if you have clear understanding on how to update it and it is not a problem for you - i'm fine with it.